src/packages/views.py

```diff
@@ -91,37 +129,65 @@ def partner_new_order(request):
         destination_street_number = request.POST.get('destination_street_number')
         destination_address = get_address(destination_city, destination_area, destination_street, destination_street_number)
 
-        rate = float(request.POST.get('rate'))
+        rate = float(request.POST.get('rate').replace('₪', ''))
         phone_number = request.POST.get('phone_number')
+        full_name = request.POST.get('full_name')
 
-        Package.objects.create(origin=origin_address, destination=destination_address, order=order, rate=rate, phone_number=phone_number)
+        Package.objects.create(origin=origin_address, destination=destination_address, order=order, rate=rate,
+                               full_name=full_name, phone_number=phone_number)
 
     context = {
         'packages': order.related_packages(),
         'order': order,
         'partner': partner,
         'rates': partner.rates.split(','),
-        'cities': City.objects.all()
+        'cities': City.objects.all(),
+        'streets': Address.objects.values_list('street_name', flat=True),
+        'names': Package.objects.exclude(full_name='').values_list('full_name', flat=True)
     }
 
-    return render(request, 'packages/new_order.html', context)
+    return render(request, 'packages/order_edit.html', context)
 
 
-def get_address(city_name, area, street, street_number):
-    city = get_city(city_name, area)
-    address = Address.objects.filter(city=city, street_name=street, street_number=street_number)
+def package_edit_view(request, partner, order, package):
+    partner = Partner.objects.get(name=partner)
+    package = Package.objects.filter(id=package)
 
-    if not address:
-        return Address.objects.create(city=city, street_name=street, street_number=street_number)
-    return address.get()
+    if not is_member(request, partner.name) and not is_staff(request):

```

Comment:
It's a bit weird that you pass `request` to `is_member`... Maybe something that is `is_member(user, group_name)`?

Reply:
It's easier, it checks if a request was made by a member, may be rename it instead?

Reply:
I still think it doesn't make logical sense to receive a `request` object in a utility that has nothing directly related to `requests`.

Reply:
ok